ghcrawler/providers/queuing/storageQueue.js

```diff
@@ -110,47 +89,21 @@ class StorageQueue {
     await this.updateVisibilityTimeout(request)
   }
 
-  updateVisibilityTimeout(request, visibilityTimeout = 0) {
-    return new Promise((resolve, reject) => {
-      // visibilityTimeout is updated to 0 to unlock/unlease the message
-      this.client.updateMessage(
-        this.queueName,
-        request._message.messageId,
-        request._message.popReceipt,
-        visibilityTimeout,
-        (error, result) => {
-          if (error) {
-            return reject(error)
-          }
-          this._log('NAKed', request._message.body)
-          resolve(this._buildMessageReceipt(result, request._message.body))
-        }
-      )
+  async updateVisibilityTimeout(request, visibilityTimeout = 0) {
+    await this.queueClient.updateMessage(request._message.messageId, request._message.popReceipt, {
+      visibilityTimeout
     })
+    this._log('NAKed', request._message.body)
   }
 
   async flush() {
-    return new Promise((resolve, reject) => {
-      this.client.deleteQueue(this.queueName, error => {
-        if (error) return reject(error)
-        this.client.createQueueIfNotExists(this.queueName, error => {
-          if (error) return reject(error)
-          resolve()
-        })
-      })
-    })
+    await this.queueClient.clearMessages()
+    this.logger.info(`Flushed all messages from ${this.queueName}`)
   }
 
   async getInfo() {
-    return new Promise(resolve => {
-      this.client.getQueueMetadata(this.queueName, (result, error) => {
-        if (error) {
-          this.logger.error(error)
-          resolve(null)
-        }
-        resolve({ count: result[0].approximateMessageCount })
-      })
-    })
+    const properties = await this.queueClient.getProperties()
+    return { count: properties.approximateMessagesCount }
```

Comment:
This likely applies in other places as well.  In the original code, error handling logged the error.  In the new code, it appears that error handling depends on how `await` handles exceptions instead of using `try` and `catch` blocks to log the error in the same way as the original.

```suggestion
    try {
      const properties = await this.queueClient.getProperties()
      return { count: properties.approximateMessagesCount }
    } catch (error) {
      this.logger.error(error)  // Log the error
      return null  // Handle the error by returning null
    }
```

Reply:
Yeah it's a good point. There was one more place like this, I've updated it as well.